lib/websql/WebSQLDatabase.js

```diff
@@ -22,12 +38,54 @@ function TransactionTask(readOnly, txnCallback, errorCallback, successCallback)
   this.successCallback = successCallback;
 }
 
-function WebSQLDatabase(dbVersion, db) {
-  this.version = dbVersion;
-  this._db = db;
-  this._txnQueue = new Queue();
-  this._running = false;
-  this._currentTask = null;
+function WebSQLDatabase(dbVersion, db, callback) {
+  var self = this;
+  self.version = dbVersion;
+  self._db = db;
+  self._txnQueue = new Queue();
+  self._running = false;
+  self._currentTask = null;
+  self._callback = callback;
+
+  self._db.exec([createInfoTable], false, function (error) {
+    /* istanbul ignore next */
+    if (error) {
+      return self._callback(null, error);
+    }
+    self._db.exec([getVersion], true, function (error, res) {
+      /* istanbul ignore next */
+      if (error) {
+        return self._callback(null, error);
+      }
+      if (res[0].rows.length === 1) {
+        var currentVersion = res[0].rows[0].value;
+        if (self.version !== '' && self.version !== currentVersion) {
+          return self._callback(null, {
+            code: 11,
+            message: 'unable to open database, version mismatch, \'' + 
+              self.version + '\' does not match the currentVersion of \'' +
+              currentVersion + '\'',
+            name: 'InvalidStateError'
+          });
```

Comment:
I tend to prefer to use real `Error` objects in user-facing APIs. In this case we can't do `class InvalidStateError extends Error` because I made the decision awhile ago to maintain ES5 compat here (for better or worse), but we can use [inherits](https://www.npmjs.com/package/inherits) to do the same thing. Something like:

```js
function InvalidStateError(message) {
  this.message = message
  this.code = 11
}
inherits(InvalidStateError, Error)
```